vm/src/builtins/list.rs

```diff
@@ -55,16 +58,20 @@ impl PyValue for PyList {
     }
 }
 
-impl<'a> BorrowValue<'a> for PyList {
-    type Borrowed = PyRwLockReadGuard<'a, Vec<PyObjectRef>>;
+impl<'a> rustpython_common::borrow::BorrowValue<'a> for PyList {
+    type Borrowed = PyMappedRwLockReadGuard<'a, [PyObjectRef]>;
 
     fn borrow_value(&'a self) -> Self::Borrowed {
-        self.elements.read()
+        self.borrow_list()
     }
 }
 
 impl PyList {
-    pub fn borrow_value_mut(&self) -> PyRwLockWriteGuard<'_, Vec<PyObjectRef>> {
+    pub fn borrow_list(&self) -> PyMappedRwLockReadGuard<'_, [PyObjectRef]> {
```

Comment:
Do we expose names of implementation or names of python interfaces? Comparing to `as_bigint` or `as_str`, I think `list` doesn't give any hint what it returns in Rust side. (slice for read and vec for write?)

Reply:
I sorta specifically didn't want to name it after a rust type, since the 2 methods return different types, and I wanted them to be the same name just with or without `_mut`. Like `borrow_slice_mut()` would be inaccurate cause it's a Vec, and `borrow_vec()` would be inaccurate cause it's a slice. I suppose `borrow_vec` could be ok, since vec is short for vector anyway, but Rust usually prefers slice over vector for that sort of thing.

Reply:
As you said, Rust code conventionally returns slice rather than read-only vec because they are practically the same and read-only vec doesn't have much mroe thing compare to slice. For the reason, I think `borrow_vec` can be just ok name. (though I think `borrow_slice` and `borrow_vec_mut` pair is better than that)

Reply:
I'll switch it to borrow_vec; I think if someone realizes they need to borrow mutably rather than immutably, it's good to be able to have a rule of "just add `_mut` at the end" (or vice versa) rather than having to remember which types have special cases.